File: kata-8-kyu/beginner-lost-without-a-map/python/beginner_lost_without_a_map_test.py
# + =====================================================================================================================
# +
# + 8 kyu - Beginner - Lost Without a Map  [ ID: 57f781872e3d8ca2a000007e ] (beginner-lost-without-a-map)
# + URL: https://www.codewars.com/kata/57f781872e3d8ca2a000007e
# + Category: REFERENCE  |  Tags: FUNDAMENTALS | ARRAYS
# +
# + =====================================================================================================================

import codewars_test as test
from beginner_lost_without_a_map import maps


@test.describe("Fixed Tests")
def fixed_tests():
    @test.it("Basic Test Cases")
    def basic_test_cases():
        tests = [[[1, 2, 3], [2, 4, 6]], [[0, 1, 2, 3, 4, 5, 6, 7, 8, 9], [0, 2, 4, 6, 8, 10, 12, 14, 16, 18]], [[], []]]

        for inp, exp in tests:
            test.assert_equals(maps(inp), exp)


@test.describe("Random Tests")
def random_tests():

    from random import randint

    reference = lambda a: [e * 2 for e in a]

    for _ in range(100):
        test_case = [randint(-1000, 1000) for i in range(randint(1, 1000))]

        @test.it(f"testing for maps({test_case})")
        def test_case():
            test.assert_equals(maps(test_case[:]), reference(test_case[:]))

File: kata-8-kyu/beginner-lost-without-a-map/javascript/beginnerLostWithoutAMap.Test.js
//+ ====================================================================================================================
//+
//+ 8 kyu - Beginner - Lost Without a Map  [ ID: 57f781872e3d8ca2a000007e ] (beginner-lost-without-a-map)
//+ URL: https://www.codewars.com/kata/57f781872e3d8ca2a000007e
//+ Category: REFERENCE  |  Tags: FUNDAMENTALS | ARRAYS
//+
//+ ====================================================================================================================

import { assert } from "vitest"
const { maps } = require("./beginnerLostWithoutAMap")

describe("Basic tests", () => {
  it("Fixed tests", () => {
    assert.deepEqual(maps([1, 2, 3]), [2, 4, 6])
    assert.deepEqual(maps([4, 1, 1, 1, 4]), [8, 2, 2, 2, 8])
    assert.deepEqual(maps([2, 2, 2, 2, 2, 2]), [4, 4, 4, 4, 4, 4])
  })

  it("Sanity test", () => {
    const a = [1, 2, 3]
    maps(a)
    assert.deepEqual(a.join("") === "123", true, "Do not mutate your input!")
  })
})

describe("Random tests", () => {
  const refMaps = (x) => x.map((x) => x * 2)
  const names = [1, 2, 3, 4, 5, 6, 7, 8, 9]
  const rnd = (n) => Math.floor(Math.random() * n)

  it("Testing for 100 random tests", () => {
    for (let i = 100; i--; ) {
      const a = Array.from({ length: rnd(7) + 2 }, () => names[rnd(names.length)])
      assert.deepEqual(maps(a.slice()), refMaps(a.slice()), `Testing for [${a}]`)
    }
  })
})
